### PR TITLE
Output more convenient metadata for bbox and SRS

### PR DESCRIPTION
**Pull request by hobu**
Output bbox information in `filter.stats`:
```
    "bbox":
    {
      "EPSG:4326":
      {
        "bbox":
        {
          "maxx": -123.0625001,
          "maxy": 44.0624972,
          "maxz": 178.73,
          "minx": -123.0749695,
          "miny": 44.0500086,
          "minz": 123.93
        }
      },
      "native":
      {
        "bbox":
        {
          "maxx": -123.0625001,
          "maxy": 44.0624972,
          "maxz": 178.73,
          "minx": -123.0749695,
          "miny": 44.0500086,
          "minz": 123.93
        }
      }
    },
```

### CODE REVIEW

File: filters/stats/StatsFilter.cpp [hunk 1]
```diff
@@ -182,6 +180,39 @@ void StatsFilter::extractMetadata()
         t.add("position", position++);
         s.extractMetadata(t);
     }
+
+    // If we have X, Y, & Z dims, output bboxes
+    auto xs = m_stats.find(Dimension::Id::X);
```
Comment (abellgithub) [inline filters/stats/StatsFilter.cpp]: Do we always want to require the Z dimension?

File: include/pdal/util/Bounds.hpp [hunk 2]
```diff
@@ -189,6 +202,10 @@ class PDAL_DLL BOX3D : private BOX2D
     BOX3D()
        { clear(); }
 
+    BOX3D(const BOX2D& box)
```
Comment (abellgithub) [inline include/pdal/util/Bounds.hpp]: We got rid of this on purpose.  It might be OK with "explicit".  Why is it here?

Comment (hobu) [inline include/pdal/util/Bounds.hpp]: I added it so I don't have to make two pdal::Polygon constructors.

File: filters/stats/StatsFilter.cpp (hunk 1 above)
Comment (hobu) [inline filters/stats/StatsFilter.cpp]: I'm 50/50 on requiring Z's. My first pass didn't require it, and my second pass did. Interested in others' thoughts on the topic...

File: include/pdal/util/Bounds.hpp (hunk 2 above)
Comment (abellgithub) [inline include/pdal/util/Bounds.hpp]: I'd recommend making two constructors.  If you need to do some common stuff, just call a function.   You could also make the ctor explicit.  You just don't want that conversion to happen in most cases.

Comment (hobu) [inline include/pdal/util/Bounds.hpp]: I do, but I'm weird I guess. I rather don't like that there's BOX2D and BOX3D at all to be honest. What we have now is more explicit, to be sure.

**Pull request merged**